src/qt/clientmodel.cpp

```diff
@@ -118,13 +122,24 @@ double ClientModel::getVerificationProgress(const CBlockIndex *tipIn) const
     return Checkpoints::GuessVerificationProgress(Params().Checkpoints(), tip);
 }
 
-void ClientModel::updateTimer()
+void ClientModel::updateTimer1()
 {
     // no locking required at this point
     // the following calls will aquire the required lock
     Q_EMIT mempoolSizeChanged(getMempoolSize(), getMempoolDynamicUsage());
     Q_EMIT bytesChanged(getTotalBytesRecv(), getTotalBytesSent());
-    Q_EMIT transactionsPerSecondChanged(getTransactionsPerSecond()); // BU:
+    Q_EMIT transactionsPerSecondChanged(getTransactionsPerSecond());
+}
+
+void ClientModel::updateTimer2()
+{
+    // Use sweep in instead of dump so that we're not writing to disk every
+    // time the timer is triggered.  If/When the client shuts down the banned
+    // list will get updated.
+    CNode::SweepBanned();
+
+    qDebug() << QString("%1: Requesting update for peer banlist").arg(__func__);
+    QMetaObject::invokeMethod(this, "updateBanlist", Qt::QueuedConnection);
```

Comment:
With the exception of the call to `SweepBanned`, this exactly duplicates the function `BannedListChanged`.  I'm not very familiar with Qt programming, but would replacing those bottom two lines with `uiInterface.BannedListChanged();` have any issues?  This is consistent with usage in other areas of the application and improves code reuse by eliminating duplicate code.